controllers/budget_controller.py

```diff
@@ -1,76 +1,168 @@
 from sqlalchemy.orm import lazyload
 
 from models.Budget import Budget
+from models.Category import Category
 from models.ParentCategory import ParentCategory
 from session import session
+from prettytable import PrettyTable
+
+global global_user_id
+
+# def menu(global_user_id):
+#     print("MENU:")
+#     print("0. Change user")
+#     print("1. Change budget")
+#     print("2. New transaction (activity)")
+#     print("3. Edit categories")
+#     print("4. Edit budget (modify budgeted amounts)")
+#     print("5. Switch the month (the billing peroid")
+#     print("6. Reports")
+#     print("7. Save and close the program")
+#     choice = input("## YOUR CHOICE: ")
+#
+#     if choice == "0":
+#         change_user()
+#     if choice == "1":
+#         change_budget(global_user_id)
+#     if choice == "2":
+#         add_transaction()
+#     if choice == "3":
+#         edit_categories()
+#     if choice == "4":
+#         edit_budget()
+#     if choice == "5":
+#         switch_month()
+#     if choice == "6":
+#         reports()
+#     if choice == "7":
+#         print("End. Program closed.")
+#         exit()
+#     else:
+#         print("\n!!! >>> WRONG CHOICE (OUT OF RANGE). TRY AGAIN...")
+#         menu()
+
+
+def add_budget(global_user_id):
+    budget_name = input("Name of new budget: ")
+    budget = Budget(name=budget_name, user_id=global_user_id)
+    session.add(budget)
+    session.commit()
 
-# GLOBAL VARIABLES
-global_user_id = None
-global_user_name = None
 
+def change_budget(global_user_id):
+    print("\nYour budgetes:")
 
-# DEFINE LOGIC HERE
+    ''' Printing the table with user's budgetes.'''
+    # Definition of PrettyTable for budgetes.
+    table_show_budgetes = PrettyTable()
+    table_show_budgetes.field_names = ["id", "name"]
 
-# Show user's budget
-def show_budget():
-    # refer to global variables inside function
-    global global_user_id
-    global global_user_name
+    # Reading data from databe and inserting into PrettyTable
+    for instance in session.query(Budget).filter(Budget.user_id == global_user_id):
+        table_show_budgetes.add_row([Budget.give_budgets(instance)[0], Budget.give_budgets(instance)[1]])
 
-    # Try to retrieve the user's budget
-    budget_instance = session.query(Budget).filter_by(user_id=global_user_id).first()
+    print(table_show_budgetes)
 
-    # If user has no budgets
-    if budget_instance is None:
-        print("\nWhoops, you don't have any budgets yet. Shall we create one?")
-        add_budget()
+    ''' Printing the budget's table that user selected.'''
```

Comment:
Add check if user has only one budget to display it immediately without asking for the ID

Reply:
Added select_budget()